src/DBFTables.jl

```diff
@@ -308,4 +353,116 @@ function Base.getproperty(dbf::Table, name::Symbol)
     return colarr
 end
 
+
+Base.write(io::IO, dbf::Table) = Base.write(io, getfield(dbf, :header), getfield(dbf, :data), 0x1a)
+Base.write(path::AbstractString, dbf::Table) = open(io -> Base.write(io, dbf), touch(path), "w")
+
+
+"Generic .dbf writer for the Tables.jl interface."
+write(path::AbstractString, tbl) = open(io -> write(io, tbl), touch(path), "w")
+
+function write(io::IO, tbl)
+    fields, records = get_field_descriptors(tbl)
+    fieldcolumns = Dict{Symbol,Int}(f.name => i for (i,f) in enumerate(fields))
+    hsize = UInt16(length(fields) * 32 + 32)
+    rsize = UInt16(sum(x -> x.length, fields)) + 1
+
+    version = 0x03
+    last_update = today()
+    incomplete = false
+    encrypted = false
+    mdx = false
+    lang_id = 0x00
+
+    h = Header(version, last_update, records, hsize, rsize, incomplete, encrypted, mdx, lang_id, fields, fieldcolumns)
+    out = Base.write(io, h)
+
+    for row in Tables.rows(tbl)
+        out += write_record(io, fields, row)
+    end
+    out += Base.write(io, 0x1a)  # EOF marker
+    return out
+end
+
+function get_field_descriptors(tbl)
+    fields = FieldDescriptor[]
+    dct = Tables.dictcolumntable(tbl)
```

Comment:
We'll want to move this up to the parent `write` definition. Call this first, then pass it into `get_field_descriptors` and use it in `write` to iterate the rows. That way we do the conversion once and re-use it when we're iterating rows. 

Also, should we make it `Tables.dictrowtable` instead? Since we're iterating rows in `write`? It should provide the same schema as `dictcolumntable`.

Reply:
We iterate rows for `write`, but we need to determine the number of bytes of each field beforehand (which requires columns).  I'll move it to the parent function.